src/app.rs

```diff
@@ -623,4 +623,14 @@ mod tests {
         assert_eq!(app.description, Some("This is a great tool.".to_string()));
         assert_eq!(app.version, Some("0.0.1".to_string()));
     }
+
+    #[test]
+    fn create_commands_through_for_loop() {
+        let app = App::new("test").flag(Flag::new("bool", FlagType::Bool));
+        vec!["dev", "prod"]
+            .into_iter()
+            .map(|a| app.clone().command(Command::new(a)));
+        app.run(vec!["test".to_string(), "prod".to_string()]);
+        assert_eq!(app.name, "test".to_string());
```

Comment:
@ChrisB9 

Thanks for the PR !
Please fix the test like this.

```suggestion
       let apps = vec!["dev", "prod"]
            .into_iter()
            .map(|a| app.clone().command(Command::new(a)));
        
        for app in apps {
            app.run(vec!["test".to_string(), "prod".to_string()]);
            assert_eq!(app.name, "test".to_string());
        }
```